pre_commit_hooks/google-java-code-format.sh

```diff
@@ -0,0 +1,18 @@
+#!/usr/bin/env sh
+export VERSION=1.9
+mkdir -p .cache
+pushd .cache
+
+JAVA_VERSION=$(java -version 2>&1 | head -1 | sed -E 's/.*version "([^"]+)"/\1/g')
+if [[ ${JAVA_VERSION} == 1.8* ]]; then
+    export VERSION=1.7
+fi
```

Comment:
if java version is 8 then use format for java 7? why?

Reply:
That's not the language version but the version fo the formatter, <=1.7 is compiled with java8, >1.8 is compiled with > java8 and can therefore no longer run in java8.

Reply:
ho I thought they would synchronize formatter versions to java versions...
Like:
* google-java-format v1.8.12 => code formatter for java 1.8
* google-java-format v1.9.2 => code formatter for java 1.9
* google-java-format v1.11.5 => code formatter for java 1.11

But ok.